base/subarray.jl

```diff
@@ -123,8 +123,10 @@ _maybe_reshape_parent(A::AbstractArray, ::NTuple{N, Bool}) where {N} = reshape(A
     view(A, inds...)
 
 Like [`getindex`](@ref), but returns a view into the parent array `A` with the
-given indices instead of making a copy.  Calling [`getindex`](@ref) or
-[`setindex!`](@ref) on the returned `SubArray` computes the
+given indices instead of making a copy if (but not only if) `A` supports `setindex!`.
+Otherwise, it may create a new object if that is a cheap operation (e.g., `UnitRange`).
+Calling [`getindex`](@ref) or
+[`setindex!`](@ref) on the returned value (usually a `SubArray`) computes the
 indices to the parent array on the fly without checking bounds.
```

Comment:
```suggestion

Some immutable parent arrays (like ranges) may choose to simply
recompute a new array in some circumstances instead of returning
a `SubArray` if doing so is efficient and provides compatible semantics.
```